src/mame/drivers/sm7238.cpp

```diff
@@ -189,28 +210,10 @@ WRITE_LINE_MEMBER(sm7238_state::write_printer_clock)
 	m_i8251prn->write_rxc(state);
 }
 
-void sm7238_state::text_memory_clear()
-{
-	int y = 0, ptr = 0;
-
-	do
-	{
-		memset(&m_p_videoram[ptr], 0x20, m_video.stride);
-		memset(&m_p_videoram[ptr + 0x1000], 0x20, m_video.stride);
-		ptr = m_p_videoram[ptr + m_video.stride + 1] | (m_p_videoram[ptr + 0x1000 + m_video.stride + 1] << 8);
-		ptr &= 0x0fff;
-		y++;
-	} while (y < 26);
-}
-
 void sm7238_state::recompute_parameters()
 {
-	rectangle visarea;
-	int horiz_pix_total = m_video.stride * 8;
-
-	visarea.set(0, horiz_pix_total - 1, 0, KSM_DISP_VERT - 1);
-	machine().first_screen()->configure(horiz_pix_total, KSM_DISP_VERT, visarea,
-		HZ_TO_ATTOSECONDS((m_video.stride == 80) ? 60 : 57.1 ));
+	machine().first_screen()->set_visible_area(m_video.stride, m_video.stride + (m_video.stride * 8) - 1,
+		KSM_VERT_START, KSM_VERT_START + KSM_DISP_VERT - 1);
```

Comment:
On the real machine, how does it implement the mode switch?  Does it change the pixel clock?  If so, does the line frequency change?  Or does it really just change the timing of horizontal blanking?

Reply:
Manual says that pixel clock changes between 12.5 and 20.625 MHz.

Reply:
In that case, shouldn't it be calling `configure` as it was previously and updating timings?  Just changing visible area would only be correct if it's only changing blanking period.  Or do the ratios somehow work out so the line frequency is identical?

Reply:
Even if the ratios work out, I think it'd be preferable to use configure(), just for the documentation value on the changing pixel clock.

Reply:
configure() isn't working for me, haven't found the reason yet.